prusti-encoder/src/encoders/ty/data.rs

```diff
@@ -242,6 +246,7 @@ impl<'vir, D: TyDatas<'vir>> StructData<'vir, D> {
         let fields = self.fields.iter().zip(other.fields.iter());
         StructData {
             data: (&self.data, &other.data),
+            inhabited: self.inhabited || other.inhabited,
```

Comment:
I think `zip` should always combine two different representations of the same type (@JonasAlaif correct me if I'm wrong), so this should rather be an assertion that the two values are equal, like the one for the number of fields at the beginning of this function.